app/scripts/models/auth_brokers/fx-desktop.js

```diff
@@ -65,6 +71,36 @@ define([
           self, options);
     },
 
+    /**
+      * Is used to ask the client about capabilities before rendering the views
+      *
+      * @returns {Promise}
+      */
+    beforeRender: function () {
+      var self = this;
+      var defer = p.defer();
+      var result = null;
+
+      self.request(self._commands.CAPABILITIES)
+        .then(function (response) {
+          if (response && response.data && _.isObject(response.data)) {
```

Comment:
is there always going to be a `response.data`? The `request` call usually [resolves the promise with the data field](https://github.com/mozilla/fxa-content-server/blob/5cf3012dc63bf38e6be97d9c2802debffa27a632/app/scripts/lib/channels/duplex.js#L176). If there is a `data` field within that, then there is `data.data` and that seems strange and we definitely have to make sure all browsers send that format.